modules/sc-mesh-secure-deployment/src/nats/comms_nats_controller.py

```diff
@@ -468,12 +480,33 @@ def __action_certificates(self, response: requests.Response) -> str:
             certificates_dict = json.loads(response.text)["payload"]["certificates"]
 
             for certificate_name, certificate in certificates_dict.items():
-                with open(
-                    f"{self.__cbm_certs_downloaded}/{certificate_name}", "wb"
-                ) as cert_file:
-                    cert_file.write(base64.b64decode(certificate))
+                if str(certificate_name).endswith(".tar.bz2"):
+                    with open(
+                        f"{self.__cbm_certs_downloaded}/{certificate_name}", "wb"
```

Comment:
`certificate_name` shall sanitized as it can contain relative paths (for example `../../../../tmp/the-legend-was-here.tar.bz2`), allowing saving the file in a different folder than intended as well as overriding existing tarfiles :game_die:

Better it would be to ignore the name given by the client and save it with a known name (to avoid handling errors due to random encodings or even emojis in the name :cry:)

Reply:
already used: 
certificate_name = os.path.basename(certificate_name)